### ISSUE TITLE
Setup tutorial - manually creating IMAP accounts not necessary

### ISSUE THREAD

**Issue by fgma** [foxcpp/maddy#339]
# Describe the bug

maddy/docs/tutorials/setting-up.md refers to maddyctl for creating IMAP accounts in the last step. This is not necessary in my setup based on the default config. IMAP accounts are created automatically. This should be updated accordingly. 

# Environment information

* maddy version: 0.4.3

**Comment by foxcpp** [comment]
Automatic IMAP account creation is a feature meant to simplify account management when an external authentication provider is used. However, currently it is triggered only after IMAP login, not on message delivery so explicitly creating an account may be a better idea to ensure the account is functional before user uses IMAP for the first time.

**Comment by fgma** [comment]
You're right. I've tested it again without logging in using IMAP before.

I'd suggest to automatically create accounts on message delivery also. This could be enabled by a flag in the configuration.

**Comment by foxcpp** [comment]
>I'd suggest to automatically create accounts on message delivery also. This could be enabled by a flag in the configuration.

Good suggestion and is definitely possible to implement (at least for some authentication providers).

**Comment by fgma** [comment]
It should be because it is somewhat unexpected to automatically create an account when accessing it using IMAP but not when delivering a message.

I've got a few other feature requests. Should I open a new issue for each of it?

**Comment by foxcpp** [comment]
> Should I open a new issue for each of it?

Yeah, that would be preferred.

**Comment by foxcpp** [comment]
Superseded by #341.
